### DISCUSSION TITLE
Separate, auto-populating, library for downloaded media

### DISCUSSION THREAD

**Discussion by SmoochSmackers** [immich-app/immich#3838] [Feature Request]
### The feature

This is a similar request to #3150. Hopefully this isn't considered a duplicate, as I try to expand on some of the ideas mentioned there, including possible starting points for implementation. 

Clearly, I'm not the only one who dislikes having their library of photos cluttered with screenshots and the images/videos they've downloaded from the internet.

This was one of my biggest problems with Google photos.

I'd love to see a feature that can rectify this. Two possibilities come to mind:

1.) The user, on the mobile app, selects the directories on their phone that they wish to be uploaded to their main library vs those they wish to be uploaded to a separate 'downloaded' or 'screenshots' library. Ideally they could choose as many, custom named libraries as they require. Possibly this could be done with some kind of auto-archived album instead of separate libraries (see #3096).

2.) The user, on the server side, sets up multiple libraries and should be able to select conditions for which an image/video file is uploaded to one library or another. For example, the user could choose categories of metadata, like 'camera model'. If there is an entry in said camera model field, it gets uploaded into the main camera roll library, if not, it gets uploaded to the 'downloaded' or 'screenshots' library. Again, possibly this could be done with an auto-archiving album instead of separate libraries.

I'm pretty new to GitHub. If I have broken any kind of protocol, please let me know and I'll do my best to rectify ASAP.

Thank you to everyone who has been working so hard on this project! This program is already amazing!

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by AngelaDMerkel** [comment]
I agree. This applies most in phone backup scenarios where there are "images" but also "photos" and you want to back up both, but without treating them as having the same value.

**Comment by jrasm91** [comment]
Could potentially be related to smart albums (some criteria adds them to an album), archiving an album (assets added to it are automatically archived).

**Comment by SmoochSmackers** [comment]
I'm genuinely curious as to why this was closed as "outdated". Maybe I missed something in one of these updates but, as far as I can tell, there is still no way for us to separate out things like downloaded content versus photos/videos we took ourselves.

While I like the feature that automatically adds uploads to albums based on their originating mobile directory, archiving an album has a lot of drawbacks and feels like a really hacky half-solution to me.

In addition, the current methods provide no way to retroactively sort downloaded/screenshots vs user generated content. Or sort content uploaded from other devices, in the way that something akin to my initially proposed solutions might.

I do appreciate all your folks' hard work! When I started hosting immich, back in 2022, I never expected the speed and reliability that I've been getting from this project, so thank you for that!

**Comment by mmomjian** [comment]
As you mentioned, #3150 is similar enough to track this for now. We are also planning to implement a Workflows feature that may make some of this more doable. Thanks.

**Reply by SmoochSmackers** [reply]
Ah ok, that makes absolute sense! Was taken off guard by "closed as outdated" event, my b.  

I am so psyched for the Workflows feature! Again, thank you for all your hard work on this project. It's people like you who make it possible for people like me to start de-googling our lives.

Next up for me to replace: Google Maps...I'm sure that'll be easy...
